contracts/validator/contract/XDCValidator.sol

```diff
@@ -218,6 +218,15 @@ contract XDCValidator {
         candidateCount = candidateCount.sub(1);
         for (uint256 i = 0; i < candidates.length; i++) {
             if (candidates[i] == _candidate) {
+                for (uint256 j = 0; j < ownerToCandidate[msg.sender].length; j++) {
+                  if (ownerToCandidate[msg.sender][j] == candidates[i]){
+                    delete ownerToCandidate[msg.sender][j];
+                  }
+                }
+                ownerToCandidate[msg.sender] = removeZeroAddresses(ownerToCandidate[msg.sender]);
```

Comment:
The below codes will save gas:

```solidity
                address[] memory cacheArray = ownerToCandidate[msg.sender];
                uint256 cacheArrayLength = cacheArray.length;
                for (uint256 j = 0; j < cacheArrayLength; j++) {
                    if (cacheArray[j] == _candidate){
                        delete cacheArray[j];
                    }
                }
                ownerToCandidate[msg.sender] = removeZeroAddresses(cacheArray);
```

Reply:
cacheArray cant change the data from ownerToCandidate[msg.sender] , it is memory ?

Reply:
delete cacheArray[j] will not change ownerToCandidate[msg.sender]